com.ibm.wala.core/src/com/ibm/wala/types/Selector.java

```diff
@@ -24,8 +27,17 @@
 
   private final Descriptor descriptor;
   
+  private final String s;
+  
+  private static final Map<String,Selector> CACHE = new HashMap<String,Selector>();
```

Comment:
Argh, this is not supposed to be in there. I put that in to debug some weird behaviour. I'll remove it.